MD/md-14/README.md

```diff
@@ -0,0 +1,75 @@
+# MD-14: Bridge Should Use A More Secure Timelock
+- **Description**: Provide formal structure for proposing non-trivial improvements to Movement.
+- **Authors**: [Primata](mailto:primata@movementlabs.xyz)
+
+## Overview
+
+The Bridge is arguably the most important feature of an L2 and lately we have had a confusing approach to it. We might want to solidify our understanding of how it is to be used and the assumptions we rely on. Off-loading the Bridge Relayer from securing timestamps might be important here in terms of security and preventing it from being an attack vector.
+
+## Definitions
+
+* Timelock - Time delta between the current block time and the end-of-lock time.
+* Bridge Relayer - formerly referred as the bridge service, it's the infrastructure utilized to have some form of synchronicity of the bridge request state. Single point of failure.
+* Initiator Contract - Contract the user utilizes to initialze the bridge and the bridge relayer completes or refunds the user. Exists both on Ethereum and Movement.
+* Counterparty Contract - Contract the user completes the bridge and the bridge relayer locks or cancels the bridge. Exists both on Ethereum and Movement.
+
+
+## Desiderata
+
+1. 
+2.  Because it has to be lower than the Initiator Contract Timelock, whenever Timelock is consumed, it is consumed as `timelock`.
+3. Definitely use timestamps for the sake of synchronicity between Ethereum and Movement.
+4. Remove dependency of the Bridge Relayer on processing the desired timelock.
+
+### D1: Initiator Contract Timelock is a constant or a modifiable state with a minimum time
+**User Journey**: Whenever Timelock is consumed, it is consumed as `timelock * 2`. If a modifiable state, it should be at least 12 hours.
```

Comment:
So, the idea here is to not relay any timestamp information in, right? 

Instead we simply assume the bridge relay has sent the time-lock information over as quickly as it can and use the timestamp from the destination chain when the relay transaction is received. 

This removes the trust on any clock in the bridge service, thus making it harder to manipulate. 

The tradeoff is perhaps that the attacker can now attack by forcing delays on the bridge. But, these could be easier to detect than lock manipulations.

Reply:
This is one reason why we need a minimum time bound. Maybe an upper bound might be necessary just so that a user couldn't have funds stuck in the bridge for type(uint).max time

Reply:
As in the difference between the reported timestamp and the timestamp at the time the relay transaction is received?

Reply:
timelock = 20
initiator receiving a bridge at time 0
counterparty lock is called at 5
initiator completes at 10
counterparty completes at 15

It would be an issue if:
iinitiator receiving a bridge at time 0
counterparty lock is called at 21
counterparty completes at 40
initiator cancels at 41

Reply:
> minimum time bound

Only makes sense when you have quite short timelocks and the actual act of calling over the network could cause them to expire incorrectly.